cosmic-usage-core/src/main/java/com/github/missioncriticalcloud/cosmic/usage/core/repositories/es/parsers/DomainsAggregationParser.java

```diff
@@ -0,0 +1,29 @@
+package com.github.missioncriticalcloud.cosmic.usage.core.repositories.es.parsers;
+
+import static com.github.missioncriticalcloud.cosmic.usage.core.utils.MetricsConstants.DOMAINS_AGGREGATION;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import io.searchbox.core.SearchResult;
+import io.searchbox.core.search.aggregation.TermsAggregation;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DomainsAggregationParser {
+
+    public List<String> parse(final SearchResult searchResult) {
+
+        if (searchResult.getTotal() == 0) {
+            return null;
```

Comment:
I'd move the empty list that is created on line 21 to right before the `if` and return it instead of `null`.